common/v2/features/Dashboard/Settings/Export/Downloader.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Button, Typography } from '@mycrypto/ui';
+import translate from 'translations';
+
+import { makeBlob } from 'utils/blob';
+
+const FullWidthButton = styled(Button)`
+  width: 100%;
+  margin-top: 1rem;
+`;
+
+const FullWidthDownloadLink = styled.a`
+  width: 100%;
+`;
+
+class Downloader extends React.Component<{ cache: string; readCache(): void }> {
+  public state = { blob: '', name: '' };
+  public componentDidMount() {
+    this.props.readCache();
```

Comment:
Aie! Hidden dependency on `readCache` to update `cache` ?
This is dangerous. Either rely on cache XOR  on the return value of readCache. Not both.

Reply:
What is XOR? I was having trouble getting the setting page to stay in sync with the rest of the cache.

Reply:
XOR means exclusive or (A or B, but not A and B)